packages/thirdweb/src/utils/any-evm/zksync/isZkSyncChain.ts

```
@@ -1,12 +1,24 @@
 import type { Chain } from "../../../chains/types.js";
+import { withCache } from "../../promise/withCache.js";
 
-export function isZkSyncChain(chain: Chain) {
-  return (
-    chain.id === 324 ||
-    chain.id === 300 ||
-    chain.id === 302 ||
-    chain.id === 11124 ||
-    chain.id === 282 || // cronos zkevm testnet
-    chain.id === 388 // cronos zkevm mainnet
+export async function isZkSyncChain(chain: Chain) {
+  const stack = await getChainStack(chain.id);
+
+  return stack === "zksync-stack";
+}
+
+async function getChainStack(chainId: number): Promise<string> {
+  return withCache(
+    async () => {
+      const res = await fetch(`https://${chainId}.rpc.thirdweb.com/stack`);
+      const data = await res.json();
+
+      if (res.ok) {
+        return data.stack;
+      }
+
+      return null;
```

Never return null, just return the .json() and let it throw here